### PR TITLE
[Feature]Accolade Badge

### PR DESCRIPTION
**Pull request by nerdlet**
## Description

Fixes [#102](https://github.com/CodeForAfrica/ui/issues/102)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Screenshots

Screenshot 2022-06-13 at 13 17 03

Screenshot 2022-06-13 at 13 16 45

Screenshot 2022-06-13 at 13 16 29

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
:+1:

---

1. #102 is about creating a single `AccoladeBadge` component. It's fine if you want to also create a complete list of them but lets keep the naming scheme similar to other components i.e. `ImpactCard` -> `ImpactCardList`, `ArticleCard` -> `ArticleCardList`, etc., implying `AccoladeBadge` -> `AccoladeBadgeList`.
2. 👇🏽 is how it looks on my end (WanaData project)
Screenshot from 2022-06-08 08-59-16

File: apps/codeforafrica/src/components/Badge/Badge.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import { Button, Typography } from "@mui/material";
+import { styled } from "@mui/material/styles";
+import PropTypes from "prop-types";
+import React from "react";
+
+import badgeIcon from "@/codeforafrica/assets/badge.svg?url";
+import NextImageButton from "@/codeforafrica/components/NextImageButton";
+
+const BadgeButtonRoot = styled(Button, {
```
[the remark this answers is not in the document]
Comment (nerdlet) [reply apps/codeforafrica/src/components/Badge/Badge.js]: @kilemensi It isn't supposed to be clickable. The button action can be disabled. I used button to support icons props such as startIcon

File: apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import React from "react";
+
+import badgeIcon from "@/codeforafrica/assets/badge.svg?url";
+
+const AccoladeBadgeRoot = styled(Grid, {
+  slot: "Root",
+})(({ theme }) => ({
+  background: theme.palette.background.main,
+  border: 0,
+  margin: "16px",
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js:14]: `Grid` uses `margin` for aligning items so setting margin on grid can cause all sorts of issues.

File: apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      <Grid item>
+        <Grid
+          container
+          flexDirection={{ xs: "row", md: "column" }}
+          alignItems={{ xs: "center", md: "flex-start" }}
+          sx={{ padding: "4.8px" }}
+        >
+          <Typography
+            variant="subtitle1"
+            sx={{ fontSize: { xs: "14px", md: "18px", margin: "0px 16px" } }}
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js:55]:
```suggestion
            variant="body1"
            sx={{ typography: { md: "body3" }, margin: "0px 16px" } }}
```
`p1` mobile, `p3` desktop.

File: apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+          >
+            {name}
+          </Typography>
+          <Typography
+            variant="subtitle1"
+            sx={{ fontSize: { xs: "12px", md: "14px" }, margin: "0px 16px" }}
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js:61]:
```suggestion
            variant="caption"
            sx={{ typography: { md: "body1" }, margin: "0px 16px" }}
```
`caption` mobile, `p1` desktop.

File: apps/codeforafrica/src/components/AccoladeBadgeList/AccoladeBadgeList.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        display: "flex",
+        flexDirection: { xs: "column", sm: "row" },
+        alignItems: "center",
+        width: "100%",
+      }}
+      ref={ref}
+      {...other}
+    >
+      {badges.map((item) => (
+        <Grid item xs={12} sm={6} md={3}>
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AccoladeBadgeList/AccoladeBadgeList.js:30]:
```suggestion
        <Grid item xs={12} sm={6} md={3} key={item.name>
```

File: apps/codeforafrica/src/components/AccoladeBadgeList/AccoladeBadgeList.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+          <AccoladeBadge name={item.name} date={item.date} />
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AccoladeBadgeList/AccoladeBadgeList.js:31]:
```suggestion
          <AccoladeBadge {...item} />
```

Unless there are props you don't want to pass.

File: apps/codeforafrica/src/pages/projects/[slug].page.js [hunk 7]
```diff
@@ -16,9 +17,18 @@ import TeamMemberCardList from "@/codeforafrica/components/TeamMemberCardList";
...
 function Index({ project, sections, ...props }) {
+  const { badges } = project;
   return (
     <Page {...props}>
       <ProjectPageHeader {...project} />
+      <Section
+        sx={{
+          marginTop: { xs: "26.6px", sm: "20px", md: "56px" },
+          marginBottom: "42px",
+        }}
```
Comment (kilemensi) [inline apps/codeforafrica/src/pages/projects/[slug].page.js:28]: Something is not right with these... the accolade badge list doesn't align well with the rest of components on the page.

**Mobile**
Screenshot from 2022-06-10 12-31-00

**Tablet**
Screenshot from 2022-06-10 12-31-38

On desktop too.

**Review by kilemensi — COMMENTED** [review]
👍🏽 ... Looking good.

-----

Spacing is still an issue in both mobile and desktop (Figma design on the top image).

File: apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    border: 0,
+  },
+}));
+
+const AccoladeBadge = React.forwardRef(function AccoladeBadge(props, ref) {
+  const { name, date, ...other } = props;
+
+  return (
+    <AccoladeBadgeRoot container ref={ref} {...other}>
+      <Grid item sx={{ padding: "5px 0px" }}>
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js:26]: What's with the padding here?

File: apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+          <Image src={badgeIcon} alt="accolade-badge-icon" layout="fill" />
+        </Box>
+      </Grid>
+      <Grid item xs={6} md={0}>
+        <Grid
+          container
+          flexDirection={{ xs: "row", md: "column" }}
+          alignItems={{ xs: "center", md: "flex-start" }}
+          justifyContent={{ xs: "space-around", md: "none" }}
+          sx={{ padding: "9.5px 0px" }}
```
Comment (kilemensi) [inline apps/codeforafrica/src/components/AccoladeBadge/AccoladeBadge.js:43]: And here?

File: apps/codeforafrica/src/pages/projects/[slug].page.js [hunk 10]
```diff
@@ -16,9 +17,20 @@ import TeamMemberCardList from "@/codeforafrica/components/TeamMemberCardList";
...
+  const { badges } = project;
   return (
     <Page {...props}>
       <ProjectPageHeader {...project} />
+      <Section
+        sx={{
+          marginTop: { xs: "26.6px", sm: "20px", md: "56px" },
+          marginBottom: "42px",
+          paddingRight: { xs: "20px", md: 0 },
+          paddingLeft: { xs: "20px", md: 0 },
```
Comment (kilemensi) [inline apps/codeforafrica/src/pages/projects/[slug].page.js:29]:
```suggestion
          px: { xs: 2.5, sm: 0 },
```

x => left and right.

**Review by kilemensi — APPROVED** [review]
🚀

**Pull request merged**